lib/plugins/acts_as_journalized/lib/redmine/acts/journalized/save_hooks.rb

```diff
@@ -79,6 +81,21 @@ def add_journal(user = User.current, notes = '')
       @journal_notes ||= notes
     end
 
+    def save_journal_with_retry(journal, tries: 2)
+      begin
+        journal.save!
+      rescue ActiveRecord::RecordInvalid => e
+        # TODO: rescue from ActiveRecord::RecordNotUnique as well
```

Comment:
`ActiveRecord::RecordNotUnique` is a wrapped database exception – I'm not sure if it actually ever gets raised.